src/containers/Statistics/StatisticsLinePlot.tsx

```diff
@@ -3,34 +3,81 @@ import LinePlotContainer from '@components/LinePlotContainer/LinePlotContainer'
 import { useDispatch, useSelector } from 'react-redux'
 import { action } from '@reducers/stats'
 import statsSelector from '@selectors/stats'
-import { getCollateralValue } from '@selectors/exchange'
 
 export const StatisticsLinePlot: React.FC = () => {
   const dispatch = useDispatch()
   const statsData = useSelector(statsSelector.linePlot)
-  const collValLastMinDay = useSelector(statsSelector.tvl)
-  const tvlValue = useSelector(getCollateralValue)
-  const [tvlData, setTvlData] = React.useState<{ value: number; percent: string }>({
+  const [timeActive, setTimeActive] = React.useState<{
+    index: number
+    serieId: string
+    timestamp: number
+    value: number
+  }>({
+    index: 0,
+    serieId: 'default',
+    timestamp: 0,
+    value: 0
+  })
+  const [menuOption, setMenuOption] = React.useState('Volume')
+  const [infoData, setInfoData] = React.useState<{ name: string; value: number; percent: string }>({
+    name: menuOption,
     value: 0,
     percent: '0'
   })
   React.useEffect(() => {
     dispatch(action.updateData())
   }, [])
 
+  const findEarlierRecord = () => {
+    return statsData.find(element => element.id === timeActive.serieId)?.points[
+      timeActive.index - 1
+    ]?.y
+  }
   React.useEffect(() => {
-    let percentTvl
-    if (collValLastMinDay.collMinDayAgo !== 0) {
-      percentTvl = (
-        ((tvlValue - collValLastMinDay.collMinDayAgo) / collValLastMinDay.collMinDayAgo) *
-        100
-      ).toFixed(2)
+    let percentTmp = '0.00'
+    let option: string
+    if (menuOption === 'User count') {
+      option = 'userCount'
     } else {
-      percentTvl = '0'
+      option = menuOption.toLowerCase()
     }
-
-    setTvlData({ ...tvlData, value: tvlValue, percent: percentTvl })
-  }, [tvlValue])
-  return <LinePlotContainer data={statsData} tvlData={tvlData} />
+    if (option !== timeActive.serieId) {
+      const tmp = statsData.find(element => element.id === option)?.points
+      if (tmp) {
+        console.log((tmp[tmp.length - 2].y * 1.0).toFixed(1))
+        if (tmp[tmp.length - 2].y * 1.0 !== 0.0) {
+          percentTmp = (
+            ((tmp[tmp.length - 1].y - tmp[tmp.length - 2].y) / tmp[tmp.length - 2].y) *
+            100
+          ).toFixed(2)
+          setInfoData({ name: menuOption, value: tmp[tmp.length - 1].y, percent: percentTmp })
+        } else {
+          percentTmp = tmp[tmp.length - 1].y.toFixed(2)
+          setInfoData({ name: menuOption, value: tmp[tmp.length - 1].y, percent: percentTmp })
+        }
+      }
+    } else {
+      const lastCol = findEarlierRecord()
+      if (lastCol) {
+        if (lastCol * 1.0 !== 0.0) {
```

Comment:
i don't thing that multiplying by one is necessary

Reply:
because tmp is an array of string and I'm converting to a floating point number